homeassistant/components/asuswrt/bridge.py

```diff
@@ -0,0 +1,289 @@
+        ret_dict[key] = values[index]
+
+    return ret_dict
+
+
+class AsusWrtBridge(ABC):
+    """The Base Bridge abstract class."""
+
+    @staticmethod
+    def get_bridge(
+        hass: HomeAssistant, conf: dict[str, Any], options: dict[str, Any] | None = None
+    ) -> AsusWrtBridge:
+        """Get Bridge instance."""
+        return AsusWrtLegacyBridge(conf, options)
+
+    def __init__(self, host: str) -> None:
+        """Initialize Bridge."""
+        self._host = host
+        self._firmware: str | None = None
+        self._label_mac: str | None = None
+        self._model: str | None = None
+
+    @property
+    def host(self) -> str:
+        """Return hostname."""
+        return self._host
+
+    @property
+    def firmware(self) -> str | None:
+        """Return firmware information."""
+        return self._firmware
+
+    @property
+    def label_mac(self) -> str | None:
+        """Return label mac information."""
+        return self._label_mac
+
+    @property
+    def model(self) -> str | None:
+        """Return model information."""
+        return self._model
+
+    @property
+    @abstractmethod
+    def is_connected(self) -> bool:
+        """Get connected status."""
+
+    @abstractmethod
+    async def async_connect(self) -> None:
+        """Connect to the device."""
+
+    @abstractmethod
+    async def async_disconnect(self) -> None:
+        """Disconnect to the device."""
+
+    @abstractmethod
+    async def async_get_connected_devices(self) -> dict[str, WrtDevice]:
+        """Get list of connected devices."""
+
+    @abstractmethod
+    async def async_get_available_sensors(self) -> dict[str, dict[str, Any]]:
+        """Return a dictionary of available sensors for this bridge."""
+
+
+class AsusWrtLegacyBridge(AsusWrtBridge):
+    """The Bridge that use legacy library."""
+
+    def __init__(
+        self, conf: dict[str, Any], options: dict[str, Any] | None = None
+    ) -> None:
+        """Initialize Bridge."""
+        super().__init__(conf[CONF_HOST])
+        self._protocol: str = conf[CONF_PROTOCOL]
+        self._api: AsusWrtLegacy = self._get_api(conf, options)
+
+    @staticmethod
+    def _get_api(
+        conf: dict[str, Any], options: dict[str, Any] | None = None
+    ) -> AsusWrtLegacy:
+        """Get the AsusWrtLegacy API."""
+        opt = options or {}
+
+        return AsusWrtLegacy(
+            conf[CONF_HOST],
+            conf.get(CONF_PORT),
+            conf[CONF_PROTOCOL] == PROTOCOL_TELNET,
+            conf[CONF_USERNAME],
+            conf.get(CONF_PASSWORD, ""),
+            conf.get(CONF_SSH_KEY, ""),
+            conf[CONF_MODE],
+            opt.get(CONF_REQUIRE_IP, True),
+            interface=opt.get(CONF_INTERFACE, DEFAULT_INTERFACE),
+            dnsmasq=opt.get(CONF_DNSMASQ, DEFAULT_DNSMASQ),
+        )
+
+    @property
+    def is_connected(self) -> bool:
+        """Get connected status."""
+        return bool(self._api.is_connected)
```

Comment:
I think `self._api.is_connected` is a bool so maybe we should use cast?
`return cast(bool, self._api.is_connected)`

Reply:
Right, changed as requested.